src/layers/activations/softmax.cu

```diff
@@ -91,44 +82,6 @@ void bp_cutoff(cudnn::cudnn_manager &cudnn,
     bp_cutoff_kernel<<<grid_dim, block_dim>>>(activations[i], error_signals[i],
                                               num_elms, min_output);
   }
-#endif
-}
-
-
-__global__ void bp_compute_cross_entropy_shortcut_kernel(const DataType *activations,
```

Comment:
Do we no longer use this code?

Reply:
This functionality has been dead for a while. It greatly expanded the complexity of the code without delivering a measurable performance boost. If we want to implement this shortcut, I think it would be more sound to support a fused softmax-cross-entropy layer.

Reply:
If we want to make it easier on the end-user, perhaps the model class can detect when a softmax layer outputs to cross entropy loss. It will then swap them out with a fused layer.